src/app/core/cache/builders/data-build.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { NormalizedObject } from '../models/normalized-object.model';
+import { CacheableObject } from '../object-cache.reducer';
+import { getRelationships } from './build-decorators';
+import { NormalizedObjectFactory } from '../models/normalized-object-factory';
+import { map, take } from 'rxjs/operators';
+import { hasValue, isNotEmpty } from '../../../shared/empty.util';
+import { PaginatedList } from '../../data/paginated-list';
+
+export function isRestDataObject(halObj: any) {
+  return isNotEmpty(halObj._links) && hasValue(halObj._links.self);
+}
+
+export function isRestPaginatedList(halObj: any) {
+  return hasValue(halObj.page) && hasValue(halObj._embedded);
+}
+
+export function isPaginatedList(halObj: any) {
+  return hasValue(halObj.page) && hasValue(halObj.pageInfo);
+}
+
+@Injectable()
+export class DataBuildService {
```

Comment:
We seem to be missing TypeDocs on this class & all the `function`s within this same file.